src/PathautoManager.php

```diff
@@ -122,15 +128,31 @@ public function __construct(ConfigFactoryInterface $config_factory, ModuleHandle
   /**
    * {@inheritdoc}
    */
-  public function createAlias($module, $op, $source, $data, $type = NULL, $langcode = LanguageInterface::LANGCODE_NOT_SPECIFIED) {
-    $config = $this->configFactory->get('pathauto.settings');
-
+  public function createAlias(EntityInterface $entity, $op) {
     // Retrieve and apply the pattern for this content type.
-    $pattern = $this->getPatternByEntity($module, $type, $langcode);
+    $pattern = $this->getPatternByEntity($entity);
+    if (empty($pattern)) {
+      // No pattern? Do nothing (otherwise we may blow away existing aliases...)
+      return NULL;
+    }
+
+    $entity_type_id = $entity->getEntityTypeId();
+    $source = $entity->urlInfo()->toString();
+    $config = $this->configFactory->get('pathauto.settings');
+    $langcode = $entity->language()->getId();
+    if ($entity->getEntityType()->getBundleEntityType()) {
+      $type = $entity->bundle();
+    }
+    else {
+      $type = NULL;
```

Comment:
Usually, drupal treats this case as $type = $entity_type_id, haven't read past this yet so that possibly doesn't make sense, just a notice..